File: backend/src/main/java/com/profroid/profroidapp/partsubdomain/presentationLayer/PartResponseModel.java
package com.profroid.profroidapp.partsubdomain.presentationLayer;

import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;

@Data
@Builder
public class PartResponseModel {
    private String partId;
    private String name;
    private String category;
    private Integer quantity;
    private BigDecimal price;
    private String supplier;
    private Integer lowStockThreshold;
    private Integer outOfStockThreshold;
    private Integer highStockThreshold;
    private String status;
    private Boolean available;
    private String imageFileId;
}

File: frontend/src/features/parts/models/PartResponseModel.ts
export interface PartResponseModel {
  partId: string;
  name: string;
  category: string;
  quantity: number;
  price: number;
  supplier: string;
  lowStockThreshold: number;
  outOfStockThreshold: number;
  highStockThreshold: number;
  status: string; // "In Stock" | "Low Stock" | "Out of Stock"
  available: boolean;
  imageFileId?: string | null;
}
